### PR TITLE
Jira fixture

### PR DESCRIPTION
**Pull request by liiight**
Added jira fixture

### To do:

- [ ] Enable passing the same args/kwargs as marker
- [ ] Test EVERYTHING
- [ ] Update docs

### CODE REVIEW

**Comment by liiight** [comment]
I kinda fucked up with the rebase... Honestly, I don't know why am I still allowed to do that 😛

**Comment by liiight** [comment]
I added a note to support passing the same kwargs as the marker (mainly `run`) but when i thinlk about it, it doesn't make sense to do so, since the fixture is used within the test.
If anything, I think some new logic can take place here, choose wether to `skip` or `fail` the test on open issue.
What do you think @lukas-bednar ?

**Comment by lukas-bednar** [comment]
>  I added a note to support passing the same kwargs as the marker (mainly run) but when i think about it, it doesn't make sense to do so, since the fixture is used within the test. Fixture should probably just return a boolean, nothing else. what do you think @lukas-bednar ?

Yes I think boolean is sufficient.
Thank you for PR!

File: pytest_jira.py [hunk 1]
```diff
@@ -435,5 +435,16 @@ def pytest_configure(config):
...
             )
+            config._jira = jira_plugin
             ok = config.pluginmanager.register(jira_plugin, "jira_plugin")
             assert ok
+
+
+@pytest.fixture
+def jira_issue(request):
+    def wrapper_jira_issue(issue_id):
+        jira_plugin = getattr(request.config, '_jira')
```
Comment (lukas-bednar) [inline pytest_jira.py]: I think you need to set this '_jira' attribute somewhere, probably in pytest_configure.

File: tests/test_jira.py [hunk 2]
```diff
@@ -704,3 +704,15 @@ def test_fail():
...
+def test_jira_fixture_run(testdir):
+    testdir.makeconftest(CONFTEST)
+    testdir.makepyfile("""
+        import pytest
+
+        def test_pass(jira_issue):
+            assert not jira_issue("ORG-1382")
+    """)
+    result = testdir.runpytest(*PLUGIN_ARGS)
+    result.assert_outcomes(1, 0, 0)
```
Comment (lukas-bednar) [inline tests/test_jira.py]: Can you please add negative test case as well?

File: pytest_jira.py (hunk 1 above)
Comment (liiight) [inline pytest_jira.py]: I did: https://github.com/rhevm-qe-automation/pytest_jira/pull/77/files#diff-0dd4c6e45b07e4db8301504abdcbc920R438

File: tests/test_jira.py (hunk 2 above)
Comment (liiight) [inline tests/test_jira.py]: sure

**Comment by liiight** [comment]
Currently the plugin returns bool if the plugin was successfully loaded (and according to the issue) status, but returns None if the plugin was not loaded.You think that's sufficient? Should we optionally raise an exception or something if the plugin isn't loaded?

I'm worried about false positives that can occur from simple jira connectivity issues.
Maybe something like:
```python
assert not jira_issue("ORG-1382", check_connection=True)
```
That will fail the test or raise and exception?

**Comment by KKoukiou** [comment]
@liiight I see what you mean by the false positive in case of the fixture returning None.
I think we should not bother about it.
Return value is very clear in the docs you provided and whoever uses it should handle the return values correctly.

**Comment by liiight** [comment]
Cool, thanks guys

**Pull request merged**